pkg/apiserver/registry/stats/multicastgroup/rest.go

```diff
@@ -0,0 +1,127 @@
+// Copyright 2022 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package multicastgroup
+
+import (
+	"context"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metatable "k8s.io/apimachinery/pkg/api/meta/table"
+	"k8s.io/apimachinery/pkg/apis/meta/internalversion"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	labels "k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/registry/rest"
+
+	statsv1alpha1 "antrea.io/antrea/pkg/apis/stats/v1alpha1"
+	"antrea.io/antrea/pkg/features"
+	"antrea.io/antrea/pkg/util/k8s"
+)
+
+type REST struct {
+	statsProvider statsProvider
+}
+
+// NewREST returns a REST object that will work against API services.
+func NewREST(p statsProvider) *REST {
+	return &REST{p}
+}
+
+var (
+	_ rest.Storage = &REST{}
+	_ rest.Scoper  = &REST{}
+	_ rest.Getter  = &REST{}
+	_ rest.Lister  = &REST{}
+)
+
+type statsProvider interface {
+	ListMulticastGroups() []statsv1alpha1.MulticastGroup
+	GetMulticastGroup(name string) (*statsv1alpha1.MulticastGroup, bool)
+}
+
+func (r *REST) New() runtime.Object {
+	return &statsv1alpha1.MulticastGroup{}
+}
+
+func (r *REST) NewList() runtime.Object {
+	return &statsv1alpha1.MulticastGroupList{}
+}
+
+func (r *REST) List(ctx context.Context, options *internalversion.ListOptions) (runtime.Object, error) {
+	if !features.DefaultFeatureGate.Enabled(features.Multicast) {
+		return &statsv1alpha1.MulticastGroup{}, nil
+	}
+	labelSelector := labels.Everything()
+	if options != nil && options.LabelSelector != nil {
+		labelSelector = options.LabelSelector
+	}
+	members := r.statsProvider.ListMulticastGroups()
+	items := make([]statsv1alpha1.MulticastGroup, 0, len(members))
+	for i := range members {
+		if labelSelector.Matches(labels.Set(members[i].Labels)) {
+			items = append(items, members[i])
+		}
+	}
+	metricList := &statsv1alpha1.MulticastGroupList{
+		Items: items,
+	}
+	return metricList, nil
+}
+
+func (r *REST) Get(ctx context.Context, name string, options *metav1.GetOptions) (runtime.Object, error) {
+	if !features.DefaultFeatureGate.Enabled(features.Multicast) {
+		return &statsv1alpha1.MulticastGroupList{}, nil
+	}
+	metric, exists := r.statsProvider.GetMulticastGroup(name)
```

Comment:
```suggestion
	group, exists := r.statsProvider.GetMulticastGroup(name)
```

Reply:
updated. I think naming multicastGroup is better